packages/NodeTypeResolver/DependencyInjection/PHPStanServicesFactory.php

```diff
@@ -49,7 +49,7 @@ public function __construct(
 
         $containerFactory = new ContainerFactory(getcwd());
         $this->container = $containerFactory->create(
-            SimpleParameterProvider::provideStringParameter(Option::CONTAINER_CACHE_DIRECTORY),
+            sys_get_temp_dir() . '/phpstan-container-for-rector',
```

Comment:
`containerCacheDirectory()` is used for user that doesn't have access to the `sys_get_temp_dir()` specific dir, and also for multi user purpose, this cause conflict, that's why it is needed, see issue:

- https://github.com/rectorphp/rector/issues/8112

Reply:
Also, by define this, user will got error:

```
 [ERROR] The file "/var/folders/dv/m3gvpzh94t59l94qz3w2dgfr0000gn/T/phpstan-container-for-rector" does not exist.  
```

The `create()` doesn't automatically create directory target, it need to be exists before.

Reply:
I see ;). We can go for defaults here And alkw override by user.

Reply:
what do you mean? I tried this and got error "phpstan-container-for-rector" directory not exist immediatelly, then if multi user is used, it will crash as can't be read by other user, see issue:

- https://github.com/rectorphp/rector/issues/8112

Reply:
Looking at https://github.com/rectorphp/rector/issues/8112, the 2 cache directory options are pretty confucing. There should be just one method to handle both.

Reply:
See what's @stefanides usage of multi user cause container crash error, using configurable container path resolve based on his experience:

https://github.com/rectorphp/rector/issues/8112#issuecomment-1664152207

for what he experienced.